pkg/cli/admin/release/extract.go

```diff
@@ -349,20 +354,42 @@ func (o *ExtractOptions) Run(ctx context.Context) error {
 		}
 	}
 
-	tarEntryCallbacks := []extract.TarEntryFunc{}
+	var versionInImageConfig string
+	opts.ImageConfigCallback = func(imageConfig *dockerv1client.DockerImageConfig) {
+		if imageConfig == nil {
+			klog.Info("Cannot retrieve the version because no image configuration is provided in the image to extract")
+			return
+		}
+		if c := imageConfig.Config; c != nil {
+			if v, ok := c.Labels["io.openshift.release"]; ok {
+				klog.V(2).Infof("Retrieved the version from image configuration in the image to extract: %s", v)
+				versionInImageConfig = v
```

Comment:
Can this callback be called multiple times, overwriting previous values in `versionInImageConfig``? If yes, can the callback be called in parallel, which would be a race?

Reply:
If I understand it correctly, `oc adm release extract` has only one release to extract and one image means one image config.

From my test, it is called only once.

Reply:
Can we encode that assumption and blew up with a panic or Fatal if `versionInImageConfig` is already set?

Reply:
The code has been moved to https://github.com/openshift/oc/pull/2050/files.
But I will keep this open because I did not address this comment and you might still think I should do it in this pull.

The concern is valid in general but it is unlikely to happen here.
The other callbacks of `extract.ExtractOptions` are not multi-thread safe either.